### DISCUSSION TITLE
How to avoid having a title for Fields?

### DISCUSSION THREAD

**Discussion by ssbarnea** [pydantic/pydantic#2495] [Question] [answered]
I observed that now given, an automatic title is given to all fields, one that based on the name. The problem is that I have 5000 fields and these generated titles are not only wrong but increase the size of the generate scheme considerably.

I tried giving `title=""` but it has no effect.  Still giving a custom value seems to set it, but I was still not able to remove it. 

AFAIK, the schema spec does not  require a `title`, so how can I avoid producing one?

**Comment by PrettyWood** [comment, marked as answer]
Hi @ssbarnea 
Currently there is no way to explicitly ask pydantic to not set the title.
In the meantime you can either overwrite `BaseModel.schema()` or set `Config.schema_extra` with a callable to remove the title.
For example
```py
from pydantic import BaseModel

def delete_title(schema):
    schema.pop('title', None)  # remove title of model
    for field_name, field_props in schema.get('properties', {}).items():
        field_props.pop('title', None)  # remove title of fields


class N(BaseModel, schema_extra=delete_title):
    y: str

class M(BaseModel, schema_extra=delete_title):
    x: str
    n: N


assert M.schema() == {
  "type": "object",
  "properties": {
    "x": {
      "type": "string"
    },
    "n": {
      "$ref": "#/definitions/N"
    }
  },
  "required": [
    "x",
    "n"
  ],
  "definitions": {
    "N": {
      "type": "object",
      "properties": {
        "y": {
          "type": "string"
        }
      },
      "required": [
        "y"
      ]
    }
  }
}
```
Hope it helps!

**Reply by dylanhogg** [reply]
This no longer works, but fear not, there is a new approach!

See: https://docs.pydantic.dev/1.10/usage/schema/#schema-customization

For the impatient (noting `def schema_extra` became `def json_schema_extra` in v2):

```python
from typing import Any
from pydantic import BaseModel


class Person(BaseModel):
    name: str
    age: int

    class Config:
        @staticmethod
        def schema_extra(schema: dict[str, Any], model: type['Person']) -> None:
            for prop in schema.get('properties', {}).values():
                prop.pop('title', None)
```
